testcases/crypto/digest_func.c

```diff
@@ -1184,13 +1184,23 @@ CK_RV do_SignVerify_HMAC(struct HMAC_TEST_SUITE_INFO * tsuite)
         /** create key object **/
         rc = create_GenericSecretKey(session, key, key_len, &h_key);
         if (rc != CKR_OK) {
+            if (rc == CKR_KEY_SIZE_RANGE) {
+                testcase_skip("keysize %d is not supported in slot %ld",
+                              (unsigned int) key_len, slot_id);
+                continue;
+            }
             testcase_error("create_GenericSecretKey rc=%s", p11_get_ckr(rc));
             goto error;
         }
 
         /** initialize signing **/
         rc = funcs->C_SignInit(session, &mech, h_key);
         if (rc != CKR_OK) {
+            if (rc == CKR_KEY_SIZE_RANGE) {
+                testcase_skip("keysize %d is not supported in slot %ld",
+                              (unsigned int) key_len, slot_id);
+                continue;
+            }
             testcase_error("C_SignInit rc=%s", p11_get_ckr(rc));
             goto error;
         }
```

Comment:
Ingo, have a look into testcases/common/common.c
There is a function check_supp_keysize() in there. This function is used by the digest testcases and checks based on the mechanis for valid key size range. I think, this is the better way. The mechanism should carry this information and so we can also check that the mechanism info entries are correct.

Reply:
The problem is that the EP11 library does not seem to report correct min/max key size requirements via C_GetMechInfo. If it would report correct values, the check_supp_keysize() would catch it already. But since it does not, it fails afterwards. Thus the additional return code checking. 

I know that's not nice, but I have no idea how to fix this otherwise. It might be considered as a bug of the EP11 library, but this needs further discussion with the EP11 people.

Reply:
Harald, are you ok with the changes now?

Reply:
Well, it hides an issue which may appear at the customer. We should make up bugzillas (or similar) to the ep11 guys. However, I would recommend to 1) add the check_supp_keysize() at all places where such a limitation may exist 2) have Ingo's workaround with skip test when the actual call fails with returncode CKR_KEY_SIZE_RANGE.

Reply:
There is already a call to check_supp_keysize()  inside do_SignVerify_HMAC(). However, it does not catch this because the EP11 library reports min/max key sizes different from what it seems to accept later on. 

Also EP11 may complain about the key size during C_CreateObject called from create_GenericSecretKey() when the key size is too small. There is no way to check the supported min/max key size, since C_CreateObject does not use/reference a mechanism. Looks like as if Ep11 simply has an overall minimum key size that it allows. This might be related to a control point setting on the EP11 card.

So yes, we should report this to the EP11 guys, but we still need a interim fix until the EP11 library is fixed (if it will be fixed at all).